### PR TITLE
fix: freeze chunks in write path (#2021)

### LINKED ISSUE #2021: Write path no longer freezes MUB
#1875 removed the [logic to freeze the MUB chunk in the write path](https://github.com/influxdata/influxdb_iox/pull/1875/files#diff-45906f4b20ff30f9970a08127270631ed8d53922585aefa8625a436827b39fddL732) on exceeding the size threshold, this now means that the freezing of MUB chunks is only triggered by the background lifecycle worker. This is unfortunate, and makes writing certain types of test challenging.

This logic should be restored, although now triggered off row threshold not size.

FYI @pauldix 

EDIT: This might have been what #1877 was created to track

### PR DESCRIPTION
**Pull request by tustvold**
Closes #2021

Once the MUB has accumulated enough data, it should be immediately frozen. I believe this was accidentally changed as part of https://github.com/influxdata/influxdb_iox/pull/1875/files#diff-45906f4b20ff30f9970a08127270631ed8d53922585aefa8625a436827b39fddL732

### CODE REVIEW

File: tests/end_to_end_cases/mod.rs [hunk 1]
```diff
@@ -1,18 +1,19 @@
-pub mod flight_api;
-pub mod http;
-pub mod management_api;
-pub mod management_cli;
-pub mod operations_api;
-pub mod operations_cli;
+mod flight_api;
```
Comment (tustvold) [inline tests/end_to_end_cases/mod.rs:1]: I'm not sure why these were pub

File: server/src/db.rs [hunk 2]
```diff
@@ -792,7 +793,13 @@ impl Db {
                                 }
                                 continue;
                             };
+
+                            let should_freeze = mb_chunk.rows() >= mub_row_threshold.get();
+
                             chunk.record_write();
+                            if should_freeze {
+                                chunk.freeze().expect("freeze mub chunk");
```
Comment (alamb) [inline server/src/db.rs:801]: Is it worth `debug!()` logging that the chunk was frozen (and the row count and the mub threshold?)

**Review by alamb — APPROVED** [review]
ooks great to me 👍

File: server/src/db.rs (hunk 2 above)
Comment (tustvold) [reply server/src/db.rs:801]: I added a debug line to freeze()

**Pull request merged**